### PR TITLE
feat: add the unified artifacts manager

### PR DESCRIPTION
**Pull request by zyy17**
## What's changed

1. Add the unified artifacts manager to manage charts and binaries(we will add plugins in the future);

2. **REMOVE** the old implementation of the artifacts manager;

Related issue: https://github.com/GreptimeTeam/gtctl/issues/151, #150.

### CODE REVIEW

File: pkg/utils/file/file.go [hunk 1]
```diff
@@ -26,10 +26,14 @@ import (
 	"path/filepath"
 )
 
-func CreateDirIfNotExists(dir string) (err error) {
-	if err := os.MkdirAll(dir, 0755); err != nil && !os.IsExist(err) {
-		return err
+// EnsureDir ensures the directory exists.
```
Comment (shawnh2) [inline pkg/utils/file/file.go:29]: we should also clarify that the function will create dir if not exist in the comment.

File: pkg/deployer/baremetal/deployer.go [hunk 2]
```diff
@@ -285,24 +287,34 @@ func (d *Deployer) ListGreptimeDBClusters(ctx context.Context, options *ListGrep
...
+		} else {
+			src, err := d.am.NewSource(artifacts.GreptimeBinName, d.config.Cluster.Artifact.Version, artifacts.ArtifactTypeBinary, false)
+			if err != nil {
+				return err
+			}
 
-	binary, err := d.am.BinaryPath(GreptimeArtifactType, d.config.Cluster.Artifact)
-	if err != nil {
-		return err
+			destDir := filepath.Join(d.baseDir, "artifacts", "binaries", artifacts.GreptimeBinName, d.config.Cluster.Artifact.Version, "pkg")
```
Comment (shawnh2) [inline pkg/deployer/baremetal/deployer.go:300]: we are still using literal strings here, but later we could abstract them into one cluster-dir-manager for bare-metal.

mentioned in #150, i will work on this one.

**Review by shawnh2 — APPROVED** [review]
lgtm

**Review by daviderli614 — APPROVED** [review]
LGTM

**Pull request merged**